cc/system_library.bzl

```diff
@@ -0,0 +1,448 @@
+BAZEL_LIB_OVERRIDE_PATHS_ENV_VAR = "BAZEL_LIB_OVERRIDE_PATHS"
+BAZEL_INCLUDE_ADDITIONAL_PATHS_ENV_VAR = "BAZEL_INCLUDE_ADDITIONAL_PATHS"
+BAZEL_INCLUDE_OVERRIDE_PATHS_ENV_VAR = "BAZEL_INCLUDE_OVERRIDE_PATHS"
+ENV_VAR_SEPARATOR = ","
+ENV_VAR_ASSIGNMENT = "="
+
+def _make_flags(array_of_strings, prefix):
+    flags = []
+    if array_of_strings:
+        for s in array_of_strings:
+            flags.append(prefix + s)
+    return " ".join(flags)
+
+def _split_env_var(repo_ctx, var_name):
+    value = repo_ctx.os.environ.get(var_name)
+    if value:
+        assignments = value.split(ENV_VAR_SEPARATOR)
+        dict = {}
+        for assignment in assignments:
+            pair = assignment.split(ENV_VAR_ASSIGNMENT)
+            if len(pair) != 2:
+                fail(
+                    "Assignments should have form 'name=value', " +
+                    "but encountered {} in env variable {}"
+                        .format(assignment, var_name),
+                )
+            key, value = pair[0], pair[1]
+            if not dict.get(key):
+                dict[key] = []
+            dict[key].append(value)
+        return dict
+    else:
+        return {}
+
+def _get_list_from_env_var(repo_ctx, var_name, key):
+    return _split_env_var(repo_ctx, var_name).get(key, default = [])
+
+def _execute_bash(repo_ctx, cmd):
+    return repo_ctx.execute(["/bin/bash", "-c", cmd]).stdout.replace("\n", "")
+
+def _find_linker(repo_ctx):
+    ld = _execute_bash(repo_ctx, "which ld")
+    lld = _execute_bash(repo_ctx, "which lld")
+    if ld:
+        return ld
+    elif lld:
+        return lld
+    else:
+        fail("No linker found")
+
+def _find_compiler(repo_ctx):
+    gcc = _execute_bash(repo_ctx, "which g++")
+    clang = _execute_bash(repo_ctx, "which clang++")
+    if gcc:
+        return gcc
+    elif clang:
+        return clang
+    else:
+        fail("No compiler found")
+
+def _find_lib_path(repo_ctx, lib_name, archive_names, lib_path_hints):
+    override_paths = _get_list_from_env_var(
+        repo_ctx,
+        BAZEL_LIB_OVERRIDE_PATHS_ENV_VAR,
+        lib_name,
+    )
+    additional_paths = _get_list_from_env_var(
+        repo_ctx,
+        BAZEL_LIB_ADDITIONAL_PATHS_ENV_VAR,
+        lib_name,
+    )
+
+    # Directories will be searched in order
+    path_flags = _make_flags(
+        override_paths + lib_path_hints + additional_paths,
+        "-L",
+    )
+    linker = _find_linker(repo_ctx)
+    for archive_name in archive_names:
+        cmd = """
+              {} -verbose -l:{} {} 2>/dev/null | \\
+              grep succeeded | \\
+              head -1 | \\
+              sed -e 's/^\\s*attempt to open //' -e 's/ succeeded\\s*$//'
+              """.format(
+            linker,
+            archive_name,
+            path_flags,
+        )
+        path = _execute_bash(repo_ctx, cmd)
+        if path:
+            return (archive_name, path)
+    return ("", "")
```

Comment:
This will be checked later to give an error right? I think it's better if you return None and check for that.